### ISSUE TITLE
Supabase Auth latest gotrue migration impossible to resync using Prisma migrate

### ISSUE THREAD

**Issue by ODreelist** [supabase/auth#1586]
## Describe the bug

We've been using supabase auth with prisma in multischema since late last year with no issues. Whenever supabase would update auth schema and prisma drift was detected I would simply do this fairly predictable process:

1.  look up the changes causing drift detection in the gotrue/migrations repo
2. create a drift_sync migration that reflects all the changes made to our supabase auth schema (will be in the gotrue/migrations repo)
3. run prisma init/prisma db pull somewhere outside the main repo to see what the changed auth schema looks like and then copy the schema.prisma changes into our main schema.prisma file
4. run prisma migrate resolve --applied 20231211214521_whatever_you_named_it

Afterwards, when you attempted to run a migrate the drift would be resolved.

However, the most recent migration: 

https://github.com/supabase/auth/blob/master/migrations/20240427152123_add_one_time_tokens_table.up.sql

Is unable to be resynced by any prisma migrate resolve order that I can figure out. Whenever I add the enum (via applied migrate resolve) prisma seems to wants to add then remove it. 

## To Reproduce

Attempt to use Supabase in multischema for any length of time and run into drift due to them changing the auth schema, then attempt to resolve it when they add an enum. 

## Expected behavior

We need a way to reliably resync our prisma schema regardless of what changes are made to the auth schema in order for multischema to be usable at all. Otherwise the promise of a usable prisma integration as outlined here:

https://supabase.com/partners/integrations/prisma

Does not hold up. 

This setup allowed us to maximize our use of Supabase in some invaluable ways.  Not only could we use Supabase Auth and RLS to its full potential we are able to document and manage RLS policies through Prisma migrations in addition to our schema, custom functions, and triggers all while making use of the auth.uid() and auth.jwt() functions inside of prisma migration.sql files. 

I'd be happy to elaborate on how this setup is achieved from a new Supabase instance and/or work with some of the devs directly because this configuration (even with the occasional drift fixing compromise) is one of the best developer experiences I have came across. I would hate to lose it because of something that is likely prisma's fault (hopefully an oversight by me) but ultimately is the Supabase community's problem to figure out.

## Screenshots

After applying the migration as is from the gotrue repo and updating the schema.prisma file to reflect the new enum and one time token table (same process that worked for all previous gotrue migrations since late december):

![image](https://github.com/supabase/auth/assets/129896797/d3ed2572-fa56-4471-b070-a04b7c26f051)

I appreciate any feedback or support, I'm not sure if we'd be able to continue with this solution if we can't get this figured out. The risk of having to blow out our DB every month is too much to assume in production.

**Comment by ODreelist** [comment]
We can leave this up if you guys want but I'm very relieved I was able to solve this issue on my own. I needed to add the schema declaration (auth.) to the enum on creation like so: 

![image](https://github.com/supabase/auth/assets/129896797/3c11d04a-72bd-40e4-bb6a-60f5a1a06d46)

in my migration.sql, otherwise it would appear it was attempting to add the enum to the default schema which would make sense when you look at the duplicate enum in the drift detection error screenshot above. 

Note: I also had to add schema declaration (auth.) the one_time_tokens.token_type column as shown here: 

![image](https://github.com/supabase/auth/assets/129896797/e0398600-1d91-4175-b341-25b86d40a9ee)

otherwise you would get this error on migrate: 

![image](https://github.com/supabase/auth/assets/129896797/4b95b65d-ffb4-4381-85c0-f3c96ba061c1)

All in all consider this issue solved and an oversight by me (best possible outcome), hopefully it helps anyone who has to keep the auth schema in sync in response to supabase making changes. You can't just copy the migrations from this repo: 

https://github.com/supabase/auth/blob/master/migrations/

change the namespace to auth and call it a day. As the changes get applied to your Supabase instances and cause prisma to detect drift, you have to carefully consider the code going in those migration.sql files and the changes you make to your schema.prisma as well. 

Again, consider this issue solved. Archive it if you feel like it can help people or let me know if you want me to delete it.

Thanks.

**Comment by hf** [comment]
We strongly recommend not using Prisma or anything else to touch anything in the `auth` schema. You can very easily break your Supabase Auth server, and we may not be able to help you recover data if you do so.

We are seeing quite a lot of problems caused with this, so in the mid-term it is very likely we will implement some measures that would prevent you from touching the `auth` schema at all. You would be able to turn this off, at your own risk.

**Comment by ODreelist** [comment]
For clarity, we don't "touch" the auth schema we reference the users.id outside of the auth schema which is supported per the docs for anyone using supabase auth.  

I do suggest not giving up on this because it does allow for a robust and efficient development team experience. I would go as far as working with the prisma to allow a schema like auth to be referenced but not "managed" in the same way prisma wants to strictly manage all included schemas. 

If I weren't able to use prisma do you have plans to create or support a first/third party tool kit primarily as a way to manage changes in a development supabase environment (db schema's, RLS policies, functions, triggers, etc.) and deploying them to a production supabase environment? 

To put it another way, If I started a new suapbase env I could have it set up in minutes thanks to the prisma schema and migrations I have in our codebase. This is the type of functionality we will be looking for.

I appreciate your feedback and look forward to your response.
